packages/web-components/src/components/ic-select/ic-select.tsx

```diff
@@ -305,9 +305,10 @@ export class Select {
       }
     }
 
-    if (this.searchable) {
+    if (this.searchable && this.value) {
+      // Only set if value not null - prevents whole input value being cleared when edited
       this.searchableSelectInputValue =
-        this.getLabelFromValue(this.currValue as string) ||
+        this.getLabelFromValue(String(this.currValue)) ||
         (this.currValue as string);
```

Comment:
Why's there two declaration for one string here? 🤔

Reply:
Turns out that this code wasn't quite redundant - it was actually part of an issue causing undesired behaviour where the `value` prop of the select can be set a value which isn't valid i.e. doesn't exist in the `options` array.

I have put the code back in and raised a separate ticket for this (#3013) because sorting it is a bit more involved than just this one line change